### DISCUSSION TITLE
Feature Request: property declaration & initialization inside constructor

### DISCUSSION THREAD

**Discussion by v-zmiycharov** [dotnet/csharplang#679] [General]
It would be nice to be able to init properties like in typescript for example:

Currently in C#: 
`class MyClass
{
 private int x;
 public MyClass(int x)
 {
  this.x = x;
 }
}`

Would be nice in C#:
`class MyClass
{
 public MyClass(private int x) {}
}`

This feature would be especially useful when working with Dependency Injection.

Best regards :)

**Comment by amis92** [comment]
Look at #39 (records). I believe it would help you in such a case too.

**Comment by artdedeco** [comment]
Also https://github.com/dotnet/csharplang/blob/master/proposals/primary-constructors.md
